src/dune_rules/modules_field_evaluator.mli

```diff
@@ -19,6 +19,15 @@ type kind =
   | Implementation of Implementation.t
   | Exe_or_normal_lib
 
+val eval_no_buildable :
+     modules:Module.Source.t Module_name.Map.t
+  -> stanza_loc:Loc.t
+  -> modules_field:Ordered_set_lang.t
+  -> modules_without_implementation:Ordered_set_lang.t
+  -> private_modules:Ordered_set_lang.t
+  -> kind:kind
+  -> Module.t Module_name.Map.t
+
```

Comment:
Do we need an entirely new function? Or can we just change the `buildable` argument to something that will work for `emit` as well.

Reply:
It was a bit messy, but I tried unifying into single function in https://github.com/ocaml/dune/pull/6349/commits/5f5c115a944131541ba2e8124cc83432d056ad7e. If you have ideas on how to leave this code more maintainable, please lmk.